JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -41,27 +45,104 @@
 public class TimeSeriesSparql {
 	// kbClient with the endpoint (triplestore/owl file) specified
 	private TripleStoreClientInterface kbClient;
-	
+
 	// Namespaces for ontology and knowledge base
 	public static final String ns_ontology = "https://www.theworldavatar.com/kg/ontotimeseries/";
 	public static final String ns_kb = "https://www.theworldavatar.com/kg/ontotimeseries/";
-	
+	public static final String ns_time = "http://www.w3.org/2006/time#";
+
 	// Prefixes
 	private static final Prefix prefix_ontology = SparqlBuilder.prefix("ts", iri(ns_ontology));
 	private static final Prefix prefix_kb = SparqlBuilder.prefix("kb", iri(ns_kb));
-	
+	private static final Prefix prefix_time = SparqlBuilder.prefix("time", iri(ns_time));
+
 	// RDF type
-    private static final Iri TimeSeries = prefix_ontology.iri("TimeSeries");
-    
-    // Relationships
-    private static final Iri hasTimeSeries = prefix_ontology.iri("hasTimeSeries");
-    private static final Iri hasRDB = prefix_ontology.iri("hasRDB");
-    private static final Iri hasTimeUnit = prefix_ontology.iri("hasTimeUnit");
-
-    // Fields for class specific exceptions
+	public static final Iri TimeSeries = prefix_ontology.iri("TimeSeries");
+	public static final Iri StepwiseCumulativeTimeSeries = prefix_ontology.iri("StepwiseCumulativeTimeSeries");
+	public static final Iri CumulativeTotalTimeSeries = prefix_ontology.iri("CumulativeTotalTimeSeries");
+	public static final Iri AverageTimeSeries = prefix_ontology.iri("AverageTimeSeries");
+	public static final Iri InstantaneousTimeSeries = prefix_ontology.iri("InstantaneousTimeSeries");
+	private static final Iri Duration = prefix_time.iri("Duration");
+
+	// Relationships
+	private static final Iri hasTimeSeries = prefix_ontology.iri("hasTimeSeries");
+	private static final Iri hasRDB = prefix_ontology.iri("hasRDB");
+	private static final Iri hasTimeUnit = prefix_ontology.iri("hasTimeUnit");
+	private static final Iri hasAveragingPeriod = prefix_ontology.iri("hasAveragingPeriod");
+	private static final Iri numericDuration = prefix_time.iri("numericDuration");
+	private static final Iri unitType = prefix_time.iri("unitType");
+
+	// Fields for class specific exceptions
 	private final String exceptionPrefix = this.getClass().getSimpleName() + ": ";
 
-    /**
+	//Time Series Type
+	public static final String Average = ns_ontology + "Average";
+	public static final String StepwiseCumulative = ns_ontology + "StepwiseCumulative";
+	public static final String CumulativeTotal = ns_ontology + "CumulativeTotal";
+	public static final String Instantaneous = ns_ontology + "Instantaneous";
```

Comment:
This is again nitpicking, a cleaner way is to have something like
public static final String AVERAGE_STRING = ns_ontology + "AverageTimeSeries";
private static final Iri AVERAGE = iri(AVERAGE_STRING);

so in your deleteTimeSeries method you should not do something like
Average + "TimeSeries" to match the type

Reply:
instead match with AVERAGE_STRING

Reply:
should we use camel case instead of AVERAGE_STRING?

There is a discrepancy:
for time series IRI its ns_ontology + "AverageTimeseries"+UUID.randomUUID();
whereas for type its AverageTimeSeries

hence one with came case and the other is not.